server/manager/src/main/java/org/apache/accumulo/manager/Manager.java

```diff
@@ -582,10 +582,11 @@ public void run() {
   }
 
   private boolean shouldCleanupMigration(TabletMetadata tabletMetadata) {
+    if (!tabletMetadata.hasMigration()) {
+      return false;
+    }
     var tableState = getContext().getTableManager().getTableState(tabletMetadata.getTableId());
     var migration = tabletMetadata.getMigration();
-    Preconditions.checkState(migration != null,
-        "This method should only be called if there is a migration");
```

Comment:
Were you seeing an IllegalStateException? Something is wrong in the current impl if so

Reply:
See the associated issue for stack traces

Reply:
I see the issue you linked... Looking now

Reply:
Regarding, "the migration was removed after the filter ran, but before the TabletMetadata was returned, by some other process," I'm not sure if that's the case or not. It's a guess because the filter is pretty simple. We likely need @keith-turner to weigh in on this one.

Reply:
I think I figured out the issue. I don't believe that any of the filters are being run on the root tablet metadata. I'm likely going to put up a different PR and close this one, but I'll leave this open for now.